src/App.tsx

```diff
@@ -241,8 +235,24 @@ const App: React.FC = () => {
       const res = await fetch(
         `${GITHUB_API_URL_PREFIX}${owner}/${repo}${GITHUB_API_TREES_CONTRIBUTORS}`
       );
+      let userDetails: object[] = [];
+      const needsDetails =
+        config.AuthorConfigs.AuthorInfo.WithEmail ||
+        config.AuthorConfigs.AuthorInfo.WithLocation ||
+        config.AuthorConfigs.AuthorInfo.WithNumberOfRepos ||
+        config.AuthorConfigs.AuthorInfo.WithTwitterUsername;
       const resJSON = await res.json();
-      console.log(resJSON);
+      if (needsDetails) {
+        for (let user in resJSON) {
```

Comment:
why for Loop instead of a map? or a Reduce?

This looks like something that can easily be done by a REDUCE btw. Play around with here so you understand it!
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce
At Facebook you are not alllowed to use anything other than MAP, Filter and Reduce in your code btw. For Loops are not acceptable

Reply:
@ralph-dev I tried map... but the problem occurs when fetching I get error:
`'await' expressions are only allowed within async functions and at the top levels of modules.`
and if I don't put await I get error :
`Property 'json' does not exist on type 'Promise<Response>'.`
when converting to json

Reply:
> why for Loop instead of a map? or a Reduce?
> 
> This looks like something that can easily be done by a REDUCE btw. Play around with here so you understand it!
> https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce
> At Facebook you are not alllowed to use anything other than MAP, Filter and Reduce in your code btw. For Loops are not acceptable

@ralph-dev Same is with Reducer